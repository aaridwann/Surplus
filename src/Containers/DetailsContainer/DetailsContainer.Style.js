import { StyleSheet } from 'react-native';
import Constants from '../../Utils/Constants/Constants';
import { scaleFont, scaleHeight, scaleWidth } from '../../Utils/Scale/Scale';

const styles = StyleSheet.create({
	headWrapper: {
		width: '100%',
		height: scaleHeight(170),
		alignItems: 'center',
		justifyContent: 'center',
		backgroundColor: Constants.Colors.LightGreenSurplus,
	},
	img: {
		borderWidth: 5,
		borderColor: 'white',
		width: 120,
		height: 120,
		backgroundColor: 'white',
		borderRadius: scaleWidth(150) / 2,
	},
	username: {
		fontSize: scaleFont(30),
		color: 'white',
	},
	sectionWrapper: {
		width: '100%',
		marginVertical: 10,
		height: scaleHeight(80),
		flexDirection: 'row',
		alignItems: 'center',
		justifyContent: 'space-between',
		paddingVertical: scaleHeight(10),
		borderBottomWidth: 1.2,
		borderBottomColor: 'gray',
	},
	sectionContainer: {
		height: '100%',
		justifyContent: 'space-between',
	},
	title: {
		fontSize: scaleFont(20),
	},
	desc: {
		fontSize: scaleFont(12),
	},
	content: {
		width: '100%',
		height: '100%',
		padding: 8,
		backgroundColor: '#EEEDE7',
	},
});

export default styles;
