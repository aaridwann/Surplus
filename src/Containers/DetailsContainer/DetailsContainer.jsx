import React from 'react';
import { Image, Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { scaleSize } from '../../Utils/Scale/Scale';
import { AntDesign } from '@expo/vector-icons';
import { get } from 'lodash';
import Constants from '../../Utils/Constants/Constants';
import { LinearGradient } from 'expo-linear-gradient';
import styles from './DetailsContainer.Style';

const _renderHead = ({ img, name }) => (
	<LinearGradient
		start={{ x: 0.2, y: 0.4 }}
		end={{ x: 1, y: 1.2 }}
		colors={[
			Constants.Colors.LightGreenSurplus,
			Constants.Colors.ExtraLightGreenSurplus,
		]}
		style={styles.headWrapper}
	>
		<Image style={styles.img} source={{ uri: img }} />
		<Text style={styles.username}>{name}</Text>
	</LinearGradient>
);

const _renderSection = ({ title, desc }) => (
	<View style={styles.sectionWrapper}>
		<View style={styles.sectionContainer}>
			<Text style={styles.title}>{title}</Text>
			<Text style={styles.desc}>{desc}</Text>
		</View>
		<AntDesign name="right" size={scaleSize(20)} color="black" />
	</View>
);

const _renderContent = (props) => (
	<View style={styles.content}>
		<_renderSection title={props.email} desc={'Original Email'} />
	</View>
);

function DetailsContainer({ route }) {
	const { params } = route;
	const img = get(params, 'picture.large', '');
	const name = get(params, 'name.first', '');

	return (
		<View>
			<_renderHead img={img} name={name} />
			<_renderContent {...params} />
		</View>
	);
}

export default DetailsContainer;

DetailsContainer.propTypes = {
	route: PropTypes.object,
	navigation: PropTypes.object,
};
