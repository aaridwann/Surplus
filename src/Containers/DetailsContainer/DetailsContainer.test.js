import React from 'react';
import { render } from '@testing-library/react-native';
import DetailsContainer from './DetailsContainer';

jest.mock('expo-linear-gradient', () => ({
	LinearGradient: 'linear',
}));

describe('DetailsContainer', () => {
	const mockRoute = {
		params: {
			picture: {
				large: 'mock-image-url',
			},
			name: {
				first: 'John',
			},
			email: 'john@example.com',
		},
	};

	test('renders correctly', () => {
		const { toJSON } = render(<DetailsContainer route={mockRoute} />);
		expect(toJSON()).toMatchSnapshot();
	});
});
