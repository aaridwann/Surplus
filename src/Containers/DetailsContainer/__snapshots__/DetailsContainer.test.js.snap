// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsContainer renders correctly 1`] = `
<View>
  <linear
    colors={
      Array [
        "#50b8a8",
        "#bbfcf2",
      ]
    }
    end={
      Object {
        "x": 1,
        "y": 1.2,
      }
    }
    start={
      Object {
        "x": 0.2,
        "y": 0.4,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#50b8a8",
        "height": 340,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "mock-image-url",
        }
      }
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "white",
          "borderRadius": 150,
          "borderWidth": 5,
          "height": 120,
          "width": 120,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 60,
        }
      }
    >
      John
    </Text>
  </linear>
  <View
    style={
      Object {
        "backgroundColor": "#EEEDE7",
        "height": "100%",
        "padding": 8,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "gray",
          "borderBottomWidth": 1.2,
          "flexDirection": "row",
          "height": 160,
          "justifyContent": "space-between",
          "marginVertical": 10,
          "paddingVertical": 20,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": "100%",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 40,
            }
          }
        >
          john@example.com
        </Text>
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Original Email
        </Text>
      </View>
    </View>
  </View>
</View>
`;
